src/transactionIntentsFactories/delegationTransactionIntentsFactory.spec.ts

```diff
@@ -0,0 +1,301 @@
+import BigNumber from "bignumber.js";
+import { Address } from "../address";
+import { DelegationTransactionIntentsFactory } from "./delegationTransactionIntentsFactory";
+import { assert } from "chai";
+import { DELEGATION_MANAGER_SC_ADDRESS } from "../constants";
+import { ValidatorPublicKey } from "@multiversx/sdk-wallet-next";
+import { SignableMessage } from "../signableMessage";
+
+describe("test delegation intents factory", function () {
+    const delegationFactory = new DelegationTransactionIntentsFactory({
+        chainID: "D",
+        minGasLimit: 50000,
+        gasLimitPerByte: 1500,
+        gasLimitStake: 5000000,
+        gasLimitUnstake: 5000000,
+        gasLimitUnbond: 5000000,
+        gasLimitCreateDelegationContract: 50000000,
+        gasLimitDelegationOperations: 1000000,
+        additionalGasLimitPerValidatorNode: 6000000,
+        additionalGasLimitForDelegationOperations: 10000000
+    });
+
+    it("should build intent for new delegation contract", async function () {
+        const sender = Address.fromBech32("erd18s6a06ktr2v6fgxv4ffhauxvptssnaqlds45qgsrucemlwc8rawq553rt2");
+        const delagationCap = new BigNumber("5000000000000000000000");
+        const serviceFee = new BigNumber(10);
```

Comment:
We can also use raw strings and numbers, and the factory methods should be able to handle them appropriately. Perhaps do this in some tests? E.g. directly pass `"5000000000000000000000"`.

Reply:
Changed this test, works as expected.